code/datums/components/deployable.dm

```diff
@@ -0,0 +1,62 @@
+/datum/component/deployable
+	var/deploy_time = 5 SECONDS //Default time it takes to deploy the item
+	var/obj/thing_to_be_deployed //What do we spawn when the item is deployed successfully?
+	var/deployed_name //For getting the name of an object for examines later on
+	var/delete_on_use = TRUE //Do we delete the item being used when the object is deployed
+
+/datum/component/deployable/Initialize(deploy_time, thing_to_be_deployed, delete_on_use)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.deploy_time = deploy_time
+	src.thing_to_be_deployed = thing_to_be_deployed
+	src.delete_on_use = delete_on_use
+
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/deploy)
+
+	var/obj/item/typecast = thing_to_be_deployed
+	deployed_name = initial(typecast.name)
+
+/datum/component/deployable/proc/examine(datum/src, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	examine_list += span_notice("It looks like it can be deployed into a [deployed_name].")
```

Comment:
```suggestion
	examine_list += span_notice("It looks like it can be deployed into \a [deployed_name].")
```

Reply:
what exactly does that do

Reply:
the \a is a text macro, it reads whats in [] and determines if it needs an "a" or "an" as some deployables may gramatically require an "an" in front of it instead

Reply:
yeah ok